docs: add additional payments documentation (#27379)
MystenLabs/sui: docs/content/onchain-finance/choose-payments-model.mdx

import { SuiGrpcClient } from '@mysten/sui/grpc';
import { Transaction } from '@mysten/sui/transactions';

const USDC = '0xdba34672e30cb065b1f93e3ab55318768fd6fef66c15942c9f7cb846e2f900e7::usdc::USDC';

const client = new SuiGrpcClient({
	network: 'mainnet',
	baseUrl: 'https://fullnode.mainnet.sui.io:443',
});

const tx = new Transaction();
tx.setSender(senderAddress);

tx.moveCall({
	target: '0x2::balance::send_funds',
	typeArguments: [USDC],
	arguments: [
		tx.balance({ type: USDC, balance: 1_000_000n }), // 1 USDC
		tx.pure.address(recipient),
	],
});

const result = await client.signAndExecuteTransaction({
	transaction: tx,
	signer: keypair,
});
```

:::caution

Gasless stablecoin transfers require a minimum transfer of 0.01, and the network deprioritizes them against fee-paying transactions when it is congested. Any other Move call in the same transaction, including a swap, a mint, or a Payment Kit call, disqualifies the transfer and the transaction then requires a normal gas payment.

:::

See [Gasless Stablecoin Transfers](/develop/transaction-payment/gasless-stablecoin-transfers) for the full allowlist and the JSON-RPC fallback.

### Sponsored transaction

> A gas station pays gas on behalf of the sender, which works with any transaction type, including Payment Kit calls and custom Move modules:

Author: ```suggestion
Sponsored transactions pay gas on behalf of the sender, which works with any transaction type:
```

<ImportContent
  source="examples/ptb-cookbook/src/sponsored.ts"
  mode="code"
  tag="sponsored"
/>

> Sponsoring from a sponsor address balance removes the gas coin entirely. Pass an empty array to `setGasPayment`, which lets the user sign first and the sponsor sign afterward, and removes the risk that a competing transaction locks the gas coin:

Author: ```suggestion
Sponsoring from an address balance removes the gas coin entirely. Pass an empty array to `setGasPayment`, which lets the user sign first and the sponsor sign afterward, and removes the risk that a competing transaction locks the gas coin:
```
